### DISCUSSION TITLE
Are there any ways to resolve a conflict with E402 and a shebang line without disabling it?

### DISCUSSION THREAD

**Discussion by jarmos-san** [astral-sh/ruff#5425] [General]
I work with a bunch of executable Python scripts and all the scripts has a shebang line (`#!/usr/bin/env Python3`). That line is followed by a "module docstring" and the module imports. But Ruff flags this usage with the [E402](https://beta.ruff.rs/docs/rules/module-import-not-at-top-of-file/) error message since it expects the modules to be placed higher up on the file. Unfortunately that isn't a possibility because without the shebang lines the scripts can potentially break.

So, are there any way around resolving this conflict without necessarily disabling the E402 message?

**Comment by charliermarsh** [comment]
Do you mind including a minimally reproducible example, just to make sure there's no misunderstanding in the issue?

**Reply by jarmos-san** [reply]
Oh my bad, we identified why we were facing this conflict on some of the scripts and not the rest. Turns out the conflicting scripts in question were written like this:

```python
#!/usr/bin/env python3

"""
Module docstring describing the functionality of the script.
"""

"""
Another docstring which should instead exists AFTER the import statements
"""

# These import statements should be in between the two docstrings instead
import argparse
import platform
import subprocess

# ...rest of the script
```

**Reply by danilovmy** [reply]
module docstring:
`A docstring is a string literal that occurs as the first statement in a module, function, class, or method definition. Such a docstring becomes the __doc__ special attribute of that object.
`
https://peps.python.org/pep-0257/#what-is-a-docstring

For module:
```
# first line in file
""" 
Another docstring which should instead exists AFTER the import statements
"""

# These import statements should be in between the two docstrings instead
import argparse
import platform
import subprocess
```
i get ast.get_docstring(module): 'Another docstring which should instead exists AFTER the import statements'

For module:
```
import argparse # first line in file
import platform
import subprocess
"""
Another docstring which should instead exists AFTER the import statements
"""
```
i get ast.get_docstring(module): ''

@charliermarsh, Following rule E402 (as this implemented now in ruff) breaks PEP-0257, and functionality of basis Python AST parser: https://docs.python.org/3/library/ast.html#ast.get_docstring, I don't use pycodestyle, but, [it seems](https://github.com/PyCQA/pycodestyle/issues/824), they implemented E402 correct.

Ask for improvements.

**Reply by ntBre** [reply]
@danilovmy Would you mind opening a separate issue? I think your suggestion is unrelated to the shebang handling.

I tried both of your snippets in the playground, and we're not flagging E402 for either one ([1](https://play.ruff.rs/12bcfe08-2e57-4025-9342-873e1dbbe77b), [2](https://play.ruff.rs/734f37eb-ef9b-41c1-b1c2-073935fd0c19)). That seems like the correct behavior to me, but I could be missing something.
